project-koku/koku
# Run cost model update in ocp on cloud summary flow

## Jira Ticket

[COST-5099](https://issues.redhat.com/browse/COST-5099)

## Description

This change will add a cost model update task to the OCP on Cloud summary flow (During cloud provider processing)

This change is in addition to https://github.com/project-koku/koku/pull/5149

## Testing

1. Checkout Branch
2. Restart Koku
3. Ingest OCP on AWS data which includes a cost model to distribute unallocated costs
4. See after we process the Cloud provider and trigger summary tasks for OCP clusters we additionally include a cost model update for that cluster.
5. See all costs are correctly distributed in the API

To save time Eva put together a dandy QE test branch up for this:
1. Checkout test branch: esebesto/reproducer_distribution_issue
2. Run the test `test_api_reproduce_infra_raw_cost_distribution_issue`
3. This will stop once data is ingested
4. Check the logs for the cost model update
5. See the API is correctly distributing the infrastructure costs

By all means try the above on main and this branch to compare.

## Release Notes

```markdown
* [COST-5099](https://issues.redhat.com/browse/COST-5099) Add cost model calculation updates during OCP on Cloud processing flow.
```
